pkg/catalog/utils/version.go

```diff
@@ -95,7 +132,7 @@ func ReleaseServerVersion(serverVersion string) bool {
 	return true
 }
 
-func LatestAvailableTemplateVersion(template *v3.CatalogTemplate) (*v32.TemplateVersionSpec, error) {
+func LatestAvailableTemplateVersion(template *v3.CatalogTemplate, clusterLister v3.ClusterLister, clusterName string) (*v32.TemplateVersionSpec, error) {
```

Comment:
This `LatestAvailableTemplateVersion` subs as a `ValidateChartCompatibility` parent ? We should at least clarify that with a comment above.

Reply:
This is technically finds the latest available compatible template version, it did the same prior but only checked Rancher version constraints.